framework/src/bundle/BundlePrivate.cpp

```diff
@@ -825,8 +825,20 @@ namespace cppmicroservices
         coreCtx->services.GetUsedByBundle(this, srs);
         for (std::vector<ServiceRegistrationBase>::const_iterator i = srs.begin(); i != srs.end(); ++i)
         {
-            auto ref = i->GetReference(std::string());
-            ref.d.Load()->UngetService(this->shared_from_this(), false);
+            // wrap in try-catch to catch failures if service is already unregistered
+            // if service is unregistered, all work in UngetService is already done by Unregister() previously
+            try
+            {
+                auto ref = i->GetReference(std::string());
+                ref.d.Load()->UngetService(this->shared_from_this(), false);
+            }
+            catch (...)
+            {
+                coreCtx->logger->Log(logservice::SeverityLevel::LOG_INFO,
```

Comment:
fixed. Other PR (for concurrent bundle start and framework stop) fixes all instances of referencing bundleid rather than symbolic name in this file, so I will just update this message rather than all in this PR.